### PR TITLE
feat: flagd flag evaluation metadata

### PR DESCRIPTION
**Pull request by Kavindu-Dodan**
## This PR

Related to flagd change at https://github.com/open-feature/flagd/pull/761

Handles evaluation metadata from flagd and fills required OF SDK field

### CODE REVIEW

File: providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java [hunk 1]
```diff
@@ -377,7 +380,7 @@ private Value convertList(com.google.protobuf.ListValue protobuf) {
     /**
      * Convert openfeature value to protobuf value.
      */
-    private com.google.protobuf.Value convertPrimitive(Value value) {
+    private static com.google.protobuf.Value convertPrimitive(Value value) {
```
Comment (thisthat) [inline providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java]: I'd push `com.google.protobuf` into the import section like it is for Struct

**Review by thisthat — APPROVED** [review]

File: providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java [hunk 2]
```diff
@@ -527,6 +535,30 @@ private static FieldDescriptor getFieldDescriptor(Message message, String name)
...
+
+        for (Map.Entry<String, com.google.protobuf.Value> entry : struct.getFieldsMap().entrySet()) {
+            if (entry.getValue().hasStringValue()) {
+                builder.addString(entry.getKey(), entry.getValue().getStringValue());
+            } else if (entry.getValue().hasBoolValue()) {
+                builder.addBoolean(entry.getKey(), entry.getValue().getBoolValue());
+            } else if (entry.getValue().hasNumberValue()) {
+                builder.addDouble(entry.getKey(), entry.getValue().getNumberValue());
+            }
+        }
```
Comment (toddbaert) [inline providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java]: I like streams but that's personal preference (and probably slower?)

**Review by toddbaert — APPROVED** [review]

**Review by beeme1mr — APPROVED** [review]

File: providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java (hunk 2 above)
Comment (Kavindu-Dodan) [reply providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java]: IMO streaming here is not needed as we simply wants to iterate over the entries. However, if there's some filtering, then streams would help.

I don't think there's any performance impact for smaller collection from either approach

File: providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java (hunk 1 above)
Comment (Kavindu-Dodan) [reply providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java]: Unfortunately no, we have conflict between `com.google.protobuf.Value` & `dev.openfeature.sdk.Value` . So the protobuf value need fully qualified name

**Comment by Kavindu-Dodan** [comment]
Merging as all conversations and suggestions are applied :)

**Pull request merged**